src/main/java/io/stargate/sgv2/jsonapi/service/projection/ProjectionPath.java

```diff
@@ -174,7 +209,7 @@ private static List<String> decode(String path) {
    */
   @Override
   public String toString() {
-    return encodeNoEscaping();
+    return segments.stream().map(ProjectionPath::encode).collect(Collectors.joining("."));
```

Comment:
I think it would be more efficient to do this differently: when constructing instance, keep hold of original, escaped "path" -- and on `toString()` just return that. Re-constructing String is quite a bit of overhead, and unnecessary.